File: programs/multi_wallet/src/contexts/admin/migrate_delegate_extension.rs
use crate::{
    state::{DelegateExtensions, DomainConfig},
    ADMIN_DOMAIN_CONFIG,
};
use anchor_lang::prelude::*;

#[derive(Accounts)]
pub struct MigrateDelegateExtension<'info> {
    #[account(
        mut,
        address = admin_domain_config.load()?.authority,
    )]
    pub authority: Signer<'info>,
    #[account(
        address = ADMIN_DOMAIN_CONFIG
    )]
    pub admin_domain_config: AccountLoader<'info, DomainConfig>,
    pub system_program: Program<'info, System>,
}

impl<'info> MigrateDelegateExtension<'info> {
    pub fn process(
        ctx: Context<'_, '_, 'info, 'info, Self>,
        api_url: String,
        member: Pubkey,
    ) -> Result<()> {
        DelegateExtensions::initialize(
            api_url,
            &member,
            ctx.remaining_accounts,
            &ctx.accounts.authority,
            &ctx.accounts.system_program,
        )?;
        Ok(())
    }
}

File: packages/wallet/src/instructions/configuration/migrateDelegateExtension.ts
import { AccountRole, type Address, type TransactionSigner } from "gill";
import { getMigrateDelegateExtensionInstruction } from "../../generated";
import { getDelegateExtensionsAddress } from "../../utils";

export async function migrateDelegateExtension({
  authority,
  apiUrl,
  member,
}: {
  authority: TransactionSigner;
  apiUrl: string;
  member: Address;
}) {
  return getMigrateDelegateExtensionInstruction({
    apiUrl,
    member,
    authority,
    remainingAccounts: [
      {
        address: await getDelegateExtensionsAddress(member),
        role: AccountRole.WRITABLE,
      },
    ],
  });
}
